2996_reflection/reflection.md

```diff
@@ -2751,6 +2783,45 @@ This allows implementing [variant](#a-simple-variant-type) without having to fur
 
 If `type_class` is a reflection of a type that already has a definition, or which is in the process of being defined, the call to `define_class` is not a constant expression.
 
+### `define_static_string`
+::: std
+```c++
+namespace std::meta {
+  consteval auto define_static_string(string_view str) -> const char *;
+  consteval auto u8define_static_string(u8string_view str) -> const char8_t *;
+}
+```
+:::
+
+Given a `string_view` or `u8string_view`, returns a pointer to a null-terminated array of characters containing `str` which has static storage duration and is usable in constant expressions.
```

Comment:
I don't think we want to require that the _source_ is null-terminated, but we do want to ensure that the _result_ is.

Also not just usable in constant expressions (which string literals are also), we want to make it clear that it's usable as a non-type template parameter. 

Maybe add something like:
```cpp
template <char const* P> struct C { };

char const msg[] = "ok";
C<msg> c1; // works
C<"nope"> c2; // ill-formed
C<define_static_string("yay")> c3; // ok
```

Reply:
> I don't think we want to require that the source is null-terminated, but we do want to ensure that the result is.

I think that's what I've worded, no?

> Also not just usable in constant expressions (which string literals are also), we want to make it clear that it's usable as a non-type template parameter.

> Maybe add something like

Both great ideas; done.

Reply:
> I think that's what I've worded, no?

In the wording section, yes. In the prose here, no.

Reply:
Hmm - reads that way to me, but I could see how it can be misread. I've rephrased slightly, let me know if it's better.